website/docs/mutation.md

```diff
@@ -0,0 +1,278 @@
+        namespaceSelector: []
+        excludedNamespaces: []
+```
+
+Note that the `applyTo` section applies to the Assign CRD only. It allows to filter resoruces by the resource GVK (group version kind).
+
+The `match` section is common to both Assign and AssignMetadata. It supports the following elements:
+- kinds - the resource kind, any of the elements listed
+- labelSelector - filters resources by resource labels listed
+- namespaces - list of allowed namespaces, only resources in listed namespaces will be mutated
+- namespaceSelector - filters resources by namespace selector
+- excludedNamespaces - list of excluded namespaces, resources in listed namespaces will not be mutated
+
+Note that the resource is not filtered if an element is not present or an empty list.
+
+#### Intent
+
+This specifies what should be changed in the resource.
+
+An example of the section is shown below:
+```yaml
+    location: "spec.containers[name:foo].imagePullPolicy"
+    parameters:
+       assign:
+        value: "Always"
+```
+
+The `location` element specifies the path to be modified.
+The `parameters.assign.value` element specifies the value to be set for the element specified in `location`. Note that the value can either be a simple string or a composite value.
+
+An example of a composite value:
+```yaml
+    location: "spec.containers[name:networking]"
+    parameters: 
+      assign: 
+        value: 
+          name: "networking"
+          imagePullPolicy: Always
+
+```
+
+The `location` element can specify either a simple subelement or an element in a list.
+For example the location `spec.containers[name:foo].imagePullPolicy` would be parsed as follows:
+- ***spec**.containers[name:foo].imagePullPolicy* - the spec element
+- *spec.**containers[name:foo]**.imagePullPolicy* - container subelement of spec. The container element is a list. Out of the list choosen, an element with the `name` element having the value `foo`.
+ - *spec.containers[name:foo].**imagePullPolicy*** - in the element from the list chosen in the previous step the element `imagePullPolicy` is chosen
+
+The yaml illustrating the above `location`:
+```yaml
+spec:
+   containers:
+   -  name:foo
+      imagePullPolicy:
+```
+Note that when the list element is empty (`spec.containers[].imagePullPolicy`), all the list element will be updated.
+
+##### Conditionals
+
+The conditions for updating the resource.
+Two types of conditions exist:
+- path tests - a resource will only be updated when a specified path exists or not
+- value tests - a resource will only be updated when the existing value is/is not contained in a list of values
+
+An example of the conditionals: 
+```yaml
+parameters:
+     pathTests:
+          - subPath: "spec.containers[name:foo]"
+            condition: MustExist
+          - subPath: spec.containers[name:foo].securityContext.capabilities
+            condition: MustNotExist
+
+      ifIn: [<lists of current values that cause the value to be modified>]
```

Comment:
This bit had to change due to the current limits of what we can do with CRDs.

Here is what the current syntax looks like:

```
spec:
   parameters:
      assignIf:
         in: [<value 1>, <value 2>, <value 3>, ...]
         notIn: [<value 1>, <value 2>, <value 3>, ...]
```